code/modules/surgery/organs/liver.dm

```diff
@@ -28,13 +28,16 @@
 			if(filterToxins)
 				//handle liver toxin filtration
 				var/toxamount
-				var/static/list/listOfToxinsInThisBitch = typesof(/datum/reagent/toxin)
-				for(var/datum/reagent/toxin/toxin in listOfToxinsInThisBitch)
-					toxamount += C.reagents.get_reagent_amount(initial(toxin.id))
-
+				var/static/list/listOfToxins = typesof(/datum/reagent/toxin)
+				for(var/I in C.reagents)

```

Comment:
`reagents` is a /datum/reagents object, you'll need to loop through `C.reagents.reagent_list`